services/auth/middleware.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package auth
+
+import (
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/web/middleware"
+)
+
+// Auth is a middleware to authenticate a web user
+func Auth(authMethod Method) func(*context.Context) {
```

Comment:
How about move them to `code.gitea.io/gitea/modules/web/middleware`?

Reply:
> Moved the router auth code from module to service because of import cycles